File: baekjoon/js/1076.js
const fs = require("fs");
const buf = fs.readFileSync(0, "utf-8").toString().trim();

const colors = buf.split("\n");

const colorMap = {
  black: [0, 1],
  brown: [1, 10],
  red: [2, 100],
  orange: [3, 1_000],
  yellow: [4, 10_000],
  green: [5, 100_000],
  blue: [6, 1_000_000],
  violet: [7, 10_000_000],
  grey: [8, 100_000_000],
  white: [9, 1_000_000_000],
};

const color1Val = colorMap[colors[0]][0] * 10;
const color2Val = colorMap[colors[1]][0];

const result = (color1Val + color2Val) * colorMap[colors[2]][1];

console.log(result);

File: baekjoon/rust/1076.rs
use std::collections::HashMap;
use std::io::{read_to_string, stdin};

struct ResistorColor {
    value: u64,
    multiple: u64,
}

fn main() {
    let buf = read_to_string(stdin()).unwrap();

    let colors: Vec<&str> = buf.split_whitespace().collect();

    let color_map = HashMap::from([
        (
            "black",
            ResistorColor {
                value: 0,
                multiple: 1,
            },
        ),
        (
            "brown",
            ResistorColor {
                value: 1,
                multiple: 10,
            },
        ),
        (
            "red",
            ResistorColor {
                value: 2,
                multiple: 100,
            },
        ),
        (
            "orange",
            ResistorColor {
                value: 3,
                multiple: 1000,
            },
        ),
        (
            "yellow",
            ResistorColor {
                value: 4,
                multiple: 10000,
            },
        ),
        (
            "green",
            ResistorColor {
                value: 5,
                multiple: 100000,
            },
        ),
        (
            "blue",
            ResistorColor {
                value: 6,
                multiple: 1000000,
            },
        ),
        (
            "violet",
            ResistorColor {
                value: 7,
                multiple: 10000000,
            },
        ),
        (
            "grey",
            ResistorColor {
                value: 8,
                multiple: 100000000,
            },
        ),
        (
            "white",
            ResistorColor {
                value: 9,
                multiple: 1000000000,
            },
        ),
    ]);

    let color1 = color_map.get(colors[0]).unwrap();
    let color2 = color_map.get(colors[1]).unwrap();
    let color3 = color_map.get(colors[2]).unwrap();

    let result = (color1.value * 10 + color2.value) * color3.multiple;

    print!("{}", result);
}
